crates/astria-core/src/protocol/transaction/v1/action/mod.rs

```diff
@@ -2622,22 +2623,30 @@ impl Protobuf for CurrencyPairsChange {
             Some(raw::currency_pairs_change::Value::Addition(raw::CurrencyPairs {
                 pairs,
             })) => {
-                let pairs = pairs
+                let pairs_len = pairs.len();
+                let parsed_pairs: IndexSet<_> = pairs
                     .into_iter()
                     .map(CurrencyPair::try_from_raw)
                     .collect::<Result<_, _>>()
                     .map_err(Self::Error::invalid_currency_pair)?;
-                Ok(Self::Addition(pairs))
+                if pairs_len != parsed_pairs.len() {
+                    return Err(Self::Error::duplicate_currency_pair());
```

Comment:
From a UX pov it'd be nice to know what the dupe was, but also no big deal

Reply:
Yeah - I considered that, but I figured the trade-off of less code was worth it since it'll only be sudo user creating these and the collection will likely be small enough that it's not too difficult to just spot the dupes.  If I was putting in handling for listing any dupes in the error, I'd want to list _all_ dupes, which is slightly more involved.